src/nvim/eval/funcs.c

```diff
@@ -952,12 +953,30 @@ static void f_cindent(typval_T *argvars, typval_T *rettv, FunPtr fptr)
     rettv->vval.v_number = -1;
 }
 
+static win_T * get_optional_window(typval_T *argvars, int idx)
+{
+  win_T *win = curwin;
+
+  if (argvars[idx].v_type != VAR_UNKNOWN) {
+    win = find_win_by_nr_or_id(&argvars[idx]);
+    if (win == NULL) {
+      EMSG(_(e_invalwindow));
+      return NULL;
+    }
+  }
+  return win;
+}
+
 /*
  * "clearmatches()" function
  */
 static void f_clearmatches(typval_T *argvars, typval_T *rettv, FunPtr fptr)
 {
-  clear_matches(curwin);
+  win_T *win = get_optional_window(argvars, 0);
+
+  if (win != NULL) {
+    clear_matches(curwin);
+  }
```

Comment:
`shouldn't it be clear_matches(win)` ?

Reply:
Thanks.  Fixed.

Reply:
the vim patch had the same issue so maybe there is a follow up patch that fixes it and other things.